### PR TITLE
adding a11y tests for high contrast

### PR DESCRIPTION
**Pull request by devaradhanm**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000 - n/a
- [x] Added relevant unit test for your changes. (`npm run test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage` - n/a
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes. - n/a

#### Description of changes
Adding high contrast tests for all the existing e2e test scenarios. refactored targetPage creation & enabling highContrast to helper functions.
we do have known a11y failures (bug in axe) in high contrast in popup. so, using snapshot for a11y results in those cases

### CODE REVIEW

File: src/tests/end-to-end/common/page.ts [hunk 1]
```diff
@@ -97,6 +97,10 @@ export class Page {
         return new URL(this.underlyingPage.target().url());
     }
 
+    public async press(key: string): Promise<void> {
```
Comment (smoralesd) [inline src/tests/end-to-end/common/page.ts]: should this be called `keypress`?

File: src/tests/end-to-end/common/setup-new-target-page.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Browser } from './browser';
+import { Page } from './page';
+
+export interface TargetPageInfo {
+    page: Page;
+    tabId: number;
+}
+export async function setupNewTargetPage(browser: Browser): Promise<TargetPageInfo> {
```
Comment (smoralesd) [inline src/tests/end-to-end/common/setup-new-target-page.ts]: I'm wondering if this should be part of `Browser`

File: src/tests/end-to-end/tests/common/settings-drop-down.test.ts [hunk 3]
```diff
@@ -3,57 +3,98 @@
...
 import { launchBrowser } from '../../common/browser-factory';
 import { CommonSelectors } from '../../common/element-identifiers/common-selectors';
+import { enableHighContrast } from '../../common/enable-high-contrast';
 import { Page } from '../../common/page';
 import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';
 
 describe('Settings Dropdown', () => {
-    let browser: Browser;
-    let targetTabId: number;
+    describe('Normal mode', () => {
```
Comment (Shobhit1) [inline src/tests/end-to-end/tests/common/settings-drop-down.test.ts]: Do we really need to wrap them in a "Settings Dropdown" describe block? Just curious to know what it offers!

Comment (devaradhanm) [reply src/tests/end-to-end/tests/common/settings-drop-down.test.ts]: it just wraps that all the tests belongs to settings dropdown

**Pull request merged**